### PR TITLE
Convert ScalarMassBDFNodeKernel to NGP

### PR DESCRIPTION
**Pull request by jrood-nrel**
I think I just need to put together a unit test for this.

### CODE REVIEW

File: unit_tests/node_kernels/UnitTestScalarMassBDFNodeKernel.C [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+#include "node_kernels/ScalarMassBDFNodeKernel.h"
+
+#ifndef KOKKOS_ENABLE_CUDA
+namespace {
+namespace bdf2_golds {
+namespace scalar_mass {
+
+static constexpr double lhs[8][8] = {
+{0.10943331816113, 0, 0, 0, 0, 0, 0, 0, },
```
Comment (sayerhs) [inline unit_tests/node_kernels/UnitTestScalarMassBDFNodeKernel.C]: I am pretty certain these gold values are incorrect, the numbers along the diagonal should be 

```
gamma1 * rhoNp1 * dualVolume * dt = 1.0 * 1.0 * 0.125 / 0.1 = 1.25
```

I think we have a bug somewhere that we need to track down.

Comment (sayerhs) [reply unit_tests/node_kernels/UnitTestScalarMassBDFNodeKernel.C]: Mixture fraction deals with `density` field differently than other equation systems. So the previous comment is incorrect.

**Comment by sayerhs** [comment]
@jrood-nrel @alanw0 I added the `lumped` version of `ScalarMassElemKernel` in commit 1ff76ff3, this matches well with the gold values you have for this NodeKernel (after accounting for node-ordering). So I feel confident that this NodeKernel has been implemented correctly.

**Comment by jrood-nrel** [comment]
Thanks for verifying this. I will merge this one then. I can fix the momentum mass next week with the diff you provided.

**Pull request merged**